### PR TITLE
API Updates to match other CRTs, Phase 1

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/software/amazon/awssdk/crt/io/SocketOptions.java [hunk 1]
```diff
@@ -76,85 +76,70 @@ int getValue() {
...
     /**
-     * Creates a new set of socket options
-     * @throws CrtRuntimeException If the system is unable to allocate space for a native socket options instance
+     * Sets the socket domain
      */
-    public SocketOptions() throws CrtRuntimeException {
-        acquireNativeHandle(socketOptionsNew());
-    }
-
+    public SocketDomain domain = SocketDomain.IPv4;
```
Comment (JonathanHenson) [inline src/main/java/software/amazon/awssdk/crt/io/SocketOptions.java]: can we at least set these to ipv6?

File: src/main/java/software/amazon/awssdk/crt/iot/AwsIotMqttConnectionConfigBuilder.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+     * 
+     * @param certPath       - Path to certificate, in PEM format
+     * @param privateKeyPath - Path to private key, in PEM format
+     */
+    public static AwsIotMqttConnectionConfigBuilder newMtlsBuilderFromPath(String certPath, String privateKeyPath) {
+        AwsIotMqttConnectionConfigBuilder builder = new AwsIotMqttConnectionConfigBuilder();
+        builder.params.port = 8883;
+        builder.tlsOptions = TlsContextOptions.createWithMtlsFromPath(certPath, privateKeyPath);
+        if (TlsContextOptions.isAlpnSupported()) {
+            builder.tlsOptions.withAlpnList("x-amzn-mqtt-ca");
```
Comment (JonathanHenson) [inline src/main/java/software/amazon/awssdk/crt/iot/AwsIotMqttConnectionConfigBuilder.java]: port needs to be 443 once you do this.

File: src/main/java/software/amazon/awssdk/crt/iot/AwsIotMqttConnectionConfigBuilder.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+     * 
+     * @param certificate - Certificate, in PEM format
+     * @param privateKey  - Private key, in PEM format
+     */
+    public static AwsIotMqttConnectionConfigBuilder newMtlsBuilder(String certificate, String privateKey) {
+        AwsIotMqttConnectionConfigBuilder builder = new AwsIotMqttConnectionConfigBuilder();
+        builder.params.port = 8883;
+        builder.tlsOptions = TlsContextOptions.createWithMtls(certificate, privateKey);
+        if (TlsContextOptions.isAlpnSupported()) {
+            builder.tlsOptions.withAlpnList("x-amzn-mqtt-ca");
```
Comment (JonathanHenson) [inline src/main/java/software/amazon/awssdk/crt/iot/AwsIotMqttConnectionConfigBuilder.java]: same here.

File: src/test/java/software/amazon/awssdk/crt/test/MqttClientConnectionFixture.java [hunk 4]
```diff
@@ -121,15 +121,15 @@ public void onConnectionInterrupted(int errorCode) {
...
+                TlsContextOptions tlsOptions = TlsContextOptions.createWithMtlsFromPath(pathToCert.toString(),
                         pathToKey.toString())) {
 
             int port = TEST_PORT;
             if (!pathToCa.toString().equals("")) {
                 tlsOptions.overrideDefaultTrustStoreFromPath(null, pathToCa.toString());
             }
             if (TlsContextOptions.isAlpnSupported()) {
-                tlsOptions.setAlpnList("x-amzn-mqtt-ca");
+                tlsOptions.withAlpnList("x-amzn-mqtt-ca");
```
Comment (JonathanHenson) [inline src/test/java/software/amazon/awssdk/crt/test/MqttClientConnectionFixture.java]: port

Comment (justinboswell) [reply src/test/java/software/amazon/awssdk/crt/test/MqttClientConnectionFixture.java]: It's set on the next line.

**Pull request merged**